### ISSUE TITLE
SurveyOf2D3Dimage

### ISSUE THREAD

**Issue by iwasakishuto** [iwasakishuto/Papers-With-Comments#12]
# Survey of 2D/3D image processing
- A survey of Object Classification and Detection based on 2D/3D data
- Author: [Xiaoke Shen](https://arxiv.org/search/cs?searchtype=author&query=Shen%2C+X)
- Submitted on Wed, 29 May 2019 (v1)

### Link
- [paper](https://arxiv.org/pdf/1905.12683.pdf)

### Summary
Authough 2D image-based systems can not provide accurate 3D location, 3D methods, such as **RGB-D** and **RGB-LiDAR** based systems can, but are more complicated.

### 2D-image based systems

<img src="https://storage.googleapis.com/groundai-web-prod/media%2Fusers%2Fuser_239115%2Fproject_369613%2Fimages%2Ftasks.jpg" alt="Main Tasks">
<center>Main Tasks</center>

#### Image Classification

|Top1 vs. network|Top1 vs. operations|
|:-:|:-:|
|![Figure.7](https://storage.googleapis.com/groundai-web-prod/media%2Fusers%2Fuser_239115%2Fproject_369613%2Fimages%2Fnetwork1.jpg)|![Figure.8](https://storage.googleapis.com/groundai-web-prod/media%2Fusers%2Fuser_239115%2Fproject_369613%2Fimages%2Fnetwork2.jpg)|

#### Object Detection
- [COCO(Common Objects in Context)](http://cocodataset.org/#home), and [PASCAL VOC(Visual Object Classes)](http://host.robots.ox.ac.uk/pascal/VOC/) are available datasets to train this algorithm.
- 2D bounding box (BB) estimation only focus on **axis-aligned boxes**, so each model estimates simple center coordinates of the bb (x,y) and the offset of the bb (W,H). That's why this 2D-based method is called **"2D center offset box encoding method".**
- There are 2 types of methods for this task, **"Two-stage methods detector"**, and **"One-stage methods"**.

##### Two-stage methods
ex) RCNN, Fast RCNN, Faster RCNN

- **RCNN (Regions with CNN features)** is the first important contribution of using DNNs to solve this task.

![RCNN system object detection system overview](https://storage.googleapis.com/groundai-web-prod/media%2Fusers%2Fuser_239115%2Fproject_369613%2Fimages%2Frcnn.jpg)

1. The region of interest (ROI) of an image is proposed by the **selective search (SS) algorithm**. The number of regions is about 2,000.
```
Selective Search:
1. Generate initial sub-segmentation, we generate many candidate     regions
2. Use greedy algorithm to recursively combine similar regions into larger ones
3. Use the generated regions to produce the final candidate region proposals
More: https://ivi.fnwi.uva.nl/isis/publications/2013/UijlingsIJCV2013/UijlingsIJCV2013.pdf
```
2. Input these ROIs to CNN to extract features.
3. Do the classification with a class-specific detection SVM.
4. Predict a new BB for the detection using a class-specific BB regressor.

As shown above, RCNN training is multi-stage pipeline (First, fine-tunes a ConvNet on object proposals using **log loss**. Then, it fits SVMs to ConvNet features using **softmax**. In the third stage, BB regressors are learned.), and inference time is long because it performs a ConvNet forward pass for each object proposal.

- **Fast R-CNN** improves the RCNN by doing ConvNet operation for the whole image and using Multi-task loss to make training stage single. This network's inputs are **"an entire image"** and **"a set of object proposals"**.

![Fast R-CNN architecture](https://lilianweng.github.io/lil-log/assets/images/fast-RCNN.png)

1. Processes the whole image with several ConvNet to produce a feature map.
2. For each object proposal, a region of interest (ROI) pooling layer extracts a fixed-length feature vector from the feature map.
3. Each feature vector is fed into a sequence of fully connected layers that finally branch into 2 sibling output layers:
    - Produces softmax prpbability estimates over K object classes plus a catch-all "background" class
    - Outputs 3 real-valued numbers for each of K object classes. Each set of 4 values encodes refined BB positions for one of the K classes.

- **Faster R-CNN** improves the Fast R-CNN by integrating the **region proposal algorithm** into the CNN model as the **region proposal network(RPN)**.

![Faster R-CNN architecture](https://lilianweng.github.io/lil-log/assets/images/faster-RCNN.png)

1. The head is used to extract the features by using CNN.
2. The RPN is used to get the region proposal.
3. Using the detection network to do the object detection.

##### One-stage methods
The YOLO systems are one-stage detection systems. They do not have a separately region proposal stage. Instead they devide the original image roughly into a S×S grid and then based on these grid cell will be used as a rough region to do the further processing.

- [YOLO](https://github.com/iwasakishuto/Papers-With-Comments/issues/2)
- [YOLOv2](https://github.com/iwasakishuto/Papers-With-Comments/issues/4)
- [YOLOv3](https://github.com/iwasakishuto/Papers-With-Comments/issues/5)

#### Semantic Segmentation
**Fully Convolutional Network (FCN)** and the **Conditional Random Field (CRF)** approach became the standard method in the semantic segmentation area.

ex) DeepLab, FCIS, and mask-RCNN

### 3D-image based systems
